### PR TITLE
Update tools.ps1/tools.sh to avoid hiding errored exit codes

### PR DESCRIPTION
**Pull request by kevinchalet**

### CODE REVIEW

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -814,7 +814,8 @@ function MSBuild-Core() {
       Write-Host "See log: $buildLog" -ForegroundColor DarkGray
     }
 
-    if ($ci) {
+    # When running on Azure Pipelines, override the returned exit code to avoid double logging.
+    if ($ci -and $env:BUILD_BUILDNUMBER -ne $null) {
```
Comment (ChadNedzlek) [inline eng/common/tools.ps1:818]: I don't think the "-ne $null" is necessary.  Powershell is pretty good at truthy checks.  That will also help if, for some reason it's like, the empty string or something.

File: samples/Mimban/Mimban.Server/Program.cs [hunk 2]
```diff
@@ -126,7 +126,7 @@ await using (var scope = app.Services.CreateAsyncScope())
     var identity = new ClaimsIdentity(
         authenticationType: TokenValidationParameters.DefaultAuthenticationType,
         nameType: Claims.Name,
-        roleType: Claims.Role);
+        roleType: Claims.Role);s
```
Comment (ChadNedzlek) [inline samples/Mimban/Mimban.Server/Program.cs]: You probably don't want an S here.

File: eng/common/tools.ps1 (hunk 1 above)
Comment (kevinchalet) [reply eng/common/tools.ps1:818]: Makes sense. I opted for `-ne $null` as it *seemed* to be the most common approach in `tools.ps1` but it's definitely something that can be changed 😄

File: samples/Mimban/Mimban.Server/Program.cs (hunk 2 above)
Comment (kevinchalet) [reply samples/Mimban/Mimban.Server/Program.cs]: Hehe, I deliberately added it (see https://github.com/openiddict/openiddict-samples/pull/216/commits/3f421c37c74f5382846dbfc82520e68ee8ba9b6b) to confirm that the build turned red on a compilation error with the fix in place (without this fix, the build stays green, which is not... ideal 😄)

File: eng/common/tools.ps1 (hunk 1 above)
Comment (ChadNedzlek) [reply eng/common/tools.ps1:818]: Ah. If it's a consistent pattern, then you can keep it. I was just worried about empty strings triggering a false positive here can the wrong behavior coming up.

**Comment by kevinchalet** [comment]
@ChadNedzlek thanks for the review! If you think it's the good approach, I'll open a PR in the Arcade repo 😃

**Pull request merged**